### DISCUSSION TITLE
Speeder pad stuck on boot

### DISCUSSION THREAD

**Discussion by Reddy3d** [Guilouz/Klipper-Flsun-Speeder-Pad#168] [Q&A] [answered]
the mainsail  works but the speeder pad hangs on boot.
I followed the guide step by step
how can I solve it?
![Desktop Screenshot 2024 04 28 - 07 23 39 49](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/82967548/6d3f829e-926d-44cc-a695-b6007e765d59)

**Comment by tfin93** [comment]
I was having a similar problem yesterday.  The issue is likely with Klipperscreen.  If you re-do the klipperscreen commands in SSH that fixed it for me.  The problem then was WFI connectivity.  The pad refused to connect to my WIFI.  I was forced to use the 1.2 image from FLSun and re-do the process.  I tried this multiple times but was never successful with installing klipperscreen even when using the -fix version.

I was able to get to a functional machine by leaving the FLsun klipperscreen installed.  This got me everything I wanted with vanilla klipper.  I'm hopeful there will be a fix for the klipperscreen install.

**Reply by Reddy3d** [reply]
I tried reinstalling klipperscreen from kiauh, it crashes at 94% on DNS search, I tried several times, nothing to do. I hope for some solution

**Comment by FP3D** [comment, marked as answer]
Bonjour même probleme quand je suis le le processus d'installation de klipperScreen, ca plante. 
Lorsque je met cette ligne la dans l'installation 1 instance : 

./KlipperScreen/scripts/KlipperScreen-install-fix.sh

Il me propose un yes ou no, et quand je met yes il me propose un x ou w, j'ai tout tenté même si je pense qu'il faut mettre Yes puis Xserver, mais il plante quand même,

pi@speeder-pad:~$ ./KlipperScreen/scripts/KlipperScreen-install-fix.sh
Checking Python version
Python 3.8.10
Install as a service? (This will enable boot to console) [Y/n]

Failing command: ['/home/pi/.KlipperScreen-env/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']

et voici le code erreur 

./KlipperScreen/scripts/KlipperScreen-install-fix.sh: line 118: /home/pi/.KlipperScreen-env/bin/activate: No such file or directory
./KlipperScreen/scripts/KlipperScreen-install-fix.sh: line 123: pip: command not found
Error: pip install exited with status code 0
Trying again with new tools...
Reading package lists... Done
Building dependency tree
Reading state information... Done
build-essential is already the newest version (12.8ubuntu1.1).
cmake is already the newest version (3.16.3-1ubuntu1.20.04.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
./KlipperScreen/scripts/KlipperScreen-install-fix.sh: line 134: pip: command not found
./KlipperScreen/scripts/KlipperScreen-install-fix.sh: line 135: pip: command not found
Unable to install dependencies, aborting install.
./KlipperScreen/scripts/KlipperScreen-install-fix.sh: line 139: deactivate: command not found

**Reply by Reddy3d** [reply]
si in effetti anche io ho notato un cambiamento in istanza non presente nella guida,X o W,ho provato tutti e due,sempre errore peccato

**Reply by FP3D** [reply]
dans la page de mon imprimante j'ai ca en notification : 
1 : Avertissement Moonraker
[update_manager] : échec du chargement de l'extension KlipperScreen : [update_manager KlipperScreen] : virtualenv non valide au chemin /home/pi/.KlipperScreen-env. Vérifiez que l'option 'virtualenv' est définie sur un chemin virtualenv valide.

2 : Avertissement Moonraker
Option de configuration non analysée « origine : https://github.com/Guilouz/KlipperScreen-Flsun-Speeder-Pad.git » détectée dans la section [update_manager KlipperScreen]. Il peut s'agir d'une option qui n'est plus disponible ou du résultat d'un module qui n'a pas pu se charger. À l'avenir, cela entraînera une erreur de démarrage.

3: Avertissement Moonraker
Option de configuration non analysée « exigences : scripts/KlipperScreen-requirements.txt » détectée dans la section [update_manager KlipperScreen]. Il peut s'agir d'une option qui n'est plus disponible ou du résultat d'un module qui n'a pas pu se charger. À l'avenir, cela entraînera une erreur de démarrage.

4 : Avertissement Moonraker
Option de configuration non analysée « system_dependencies : scripts/system-dependencies.json » détectée dans la section [update_manager KlipperScreen]. Il peut s'agir d'une option qui n'est plus disponible ou du résultat d'un module qui n'a pas pu se charger. À l'avenir, cela entraînera une erreur de démarrage. 

5: Avertissement Moonraker
Option de configuration non analysée « managed_services : KlipperScreen » détectée dans la section [update_manager KlipperScreen]. Il peut s'agir d'une option qui n'est plus disponible ou du résultat d'un module qui n'a pas pu se charger. À l'avenir, cela entraînera une erreur de démarrage.

**Reply by Reddy3d** [reply]
they did a great job with this guide, we need an update on the procedure, let's hope they find a solution soon

**Reply by Guilouz** [reply]
There is some changes in KlipperScreen install with new PR, retry it.

**Reply by FP3D** [reply]
Bonjour, merci pour les corrections.

lors de l'installation de KlipperScreen, que faut-il répondre apres avoir inséré la ligne suivante ?

./KlipperScreen/scripts/KlipperScreen-install-fix.sh

Checking Python version
Python 3.8.10
Install standalone?
it will create a service, enable boot to console and install the graphical dependencies.
Say no to install as a regular desktop app that will not start automatically

Install standalone? Press enter for default (standalone) [Y/n]

Si Yes 👍 

Choose graphical backend
Default is Xserver
Wayland is EXPERIMENTAL needs kms/drm drivers doesn't support DPMS and may need autologin

Backend Xserver or Wayland (cage), Press enter for default (Xserver)? [X/w]

il faut faire Y et X ?

**Reply by FP3D** [reply]
à la fin, en tapant Y, puis X, il installe et à la fin j'ai ce message, la derniere ligne etant en rouge

Installed Misc packages
Creating virtual environment
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt install python3.8-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/home/pi/.KlipperScreen-env/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']

./KlipperScreen/scripts/KlipperScreen-install-fix.sh: line 119: /home/pi/.KlipperScreen-env/bin/activate: No such file or directory
Could not activate the enviroment, try deleting /home/pi/.KlipperScreen-env and retry

**Reply by Guilouz** [reply]
Delete /home/pi/.KlipperScreen-env folder

Then in SSH enter this command: 

sudo apt install python3.8-venv

Then run again the installer, your choice are good

**Reply by FP3D** [reply]
J'ai tout refais depuis le debut, j'ai fait la manip que tu as donné avant de lancer la procedure pour KlipperScreen, puis j'ai relancé la commande. ca a duréé plus longtemps et j'ai eu ce message.
![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/51458889/37ac7869-466b-4975-a646-bd203f20e4ca)

**Reply by FP3D** [reply]
Merci pour ton aide @Guilouz

**Reply by Guilouz** [reply]
Est-ce que ça fonctionne ?

**Reply by FP3D** [reply]
Malgré les lignes en rouge, j'ai continué le processus, et ca à l'air de fonctionner, je suis a l'etape de calibration de l'imprimante, donc ok pour moi, merci

**Reply by Guilouz** [reply]
Nice, j'ai mis à jour le wiki en conséquence.

**Reply by Reddy3d** [reply]
It works now, thank you very much.
I just have this error to resolve, any help?
![Immagine WhatsApp 2024-05-02 ore 19 44 42_47df7d3c](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/82967548/3b9499fd-0a9d-4be1-973d-8f56a39e5005)

**Reply by Guilouz** [reply]
That’s means, your moonraker.conf file is missing or corrupted and backup file is used.

You must replace moonraker.conf by mine in your config folder.

**Reply by Reddy3d** [reply]
there are no moonraker files in the folder, where can I find it?

**Reply by Guilouz** [reply]
That means you have not followed the wiki^^

https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Use-Configurations

**Reply by Reddy3d** [reply]
![Desktop Screenshot 2024 05 02 - 21 29 43 58](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/82967548/c8f9ca48-747b-4d30-b72f-ab8e648c2ddf)
sorry, I unzipped the folder and copied the files as described, but there is no moonraker.conf file in the folder.
Either I'm wrong or I didn't understand which file to replace

**Reply by Reddy3d** [reply]
I found the error, solved it. thanks to the availability
I added "#"error disappeared
![Desktop Screenshot 2024 05 02 - 21 34 46 43](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/82967548/775e2372-fd76-494a-8c86-662f5106ddf7)

**Reply by Guilouz** [reply]
Ah yes sorry, moonraker.conf is generated automatically

**Reply by Reddy3d** [reply]
I've been following you for a long time and I follow your guides precisely, thank you very much.
a question, for flsun v400 the best slicer?

**Reply by Guilouz** [reply]
Orca

**Reply by Reddy3d** [reply]
I'll try it, where is it?

**Comment by Guilouz** [comment]
- Updated the following sections to build venv with latest Python3/PIP versions to avoid install issues and to add support for latest version of KlipperScreen with NetworkManager:
    - [Update Dependencies](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Update-dependencies)
    - [Delete Flsun Builds](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Delete-Flsun-Builds)
    - [Install Official Builds (1 instance)](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Install-Official-Builds-(1-instance))
    - [Install Official Builds (Multiple instances)](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Install-Official-Builds-(Multiple-instances))

> [!CAUTION]  
> **Speeder Pad need to be restored with this [Restore OS Image File](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Restore-OS-Image-File) section for that and the entire process needs to be redone.<br />Don't forget to backup your Klipper/Moonraker configuration files first.**

![IMG_0622](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/12702322/5068a240-e811-4ea6-bcdb-241d07503d7f)
![IMG_0624](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/12702322/2d91a881-0036-4fe0-9f1d-774bea275a3f)
